### DISCUSSION TITLE
More In-Depth Option Data

### DISCUSSION THREAD

**Discussion by alvestoradmin** [ranaroussi/yfinance#2078] [Ideas]
Yahoo Finance includes historical data for individual options contracts just as they do for stocks. Maybe not as thorough, but the data is there & usable. I was trying to use the API to automate the creation of HV vs. IV comparison charts, as they can say a lot about what the market has priced in, and there wasn't even a detour I could've taken to make it work. 
<img width="1044" alt="Screenshot 2024-10-07 at 6 48 35 AM" src="https://github.com/user-attachments/assets/b1ffe807-7e27-49dd-8b1a-cd9dfc154244">
<img width="1047" alt="Screenshot 2024-10-07 at 6 49 00 AM" src="https://github.com/user-attachments/assets/e96d460e-453e-42ed-8ff8-7efe779d316a">

**Comment by alvestoradmin** [comment]
Just an add-on here, I thought I should mention that for more complex modeling even the small amount of data shown here can make a difference. This is the single best free API I've come across; this is the only shortcoming I've noticed.

**Comment by ericpien** [comment]
is there anything you need that's not available via:

```Python
import yfinance as yf
op = yf.Ticker('NVDA241011C00050000')
op.history()
```

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Open</th>
      <th>High</th>
      <th>Low</th>
      <th>Close</th>
      <th>Volume</th>
      <th>Dividends</th>
      <th>Stock Splits</th>
    </tr>
    <tr>
      <th>Date</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2024-09-09 00:00:00-04:00</th>
      <td>54.150002</td>
      <td>57.099998</td>
      <td>54.150002</td>
      <td>57.099998</td>
      <td>2</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-11 00:00:00-04:00</th>
      <td>65.599998</td>
      <td>65.599998</td>
      <td>65.599998</td>
      <td>65.599998</td>
      <td>2</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-12 00:00:00-04:00</th>
      <td>69.300003</td>
      <td>69.300003</td>
      <td>69.300003</td>
      <td>69.300003</td>
      <td>2</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-16 00:00:00-04:00</th>
      <td>66.470001</td>
      <td>66.470001</td>
      <td>66.470001</td>
      <td>66.470001</td>
      <td>1</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-18 00:00:00-04:00</th>
      <td>65.449997</td>
      <td>65.449997</td>
      <td>63.889999</td>
      <td>63.889999</td>
      <td>47</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-24 00:00:00-04:00</th>
      <td>66.550003</td>
      <td>66.550003</td>
      <td>65.660004</td>
      <td>65.660004</td>
      <td>2</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-09-30 00:00:00-04:00</th>
      <td>68.949997</td>
      <td>68.949997</td>
      <td>68.949997</td>
      <td>68.949997</td>
      <td>1</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-10-04 00:00:00-04:00</th>
      <td>74.529999</td>
      <td>74.930000</td>
      <td>73.000000</td>
      <td>74.930000</td>
      <td>43</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2024-10-07 00:00:00-04:00</th>
      <td>75.000000</td>
      <td>79.099998</td>
      <td>75.000000</td>
      <td>79.099998</td>
      <td>6</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
  </tbody>
</table>
</div>

**Reply by alvestoradmin** [reply]
I see. I didn't even think to try that. I guess I assumed that yf.Ticker() was only for non-derivative asset symbols. Thanks!

**Comment by VISHALGIT-20** [comment]
can i get option chain contract historical data  (EOD DAILY OHLC) for comex commodity for free
if there anything please let me know
